patches/tModLoader/Terraria/Graphics/Renderers/LegacyPlayerRenderer.cs.patch

```diff
@@ -0,0 +1,219 @@
++		}
++
++		private void DrawPlayerInternal(Camera camera, Player drawPlayer, Vector2 position, float rotation, Vector2 rotationOrigin, float shadow = 0f, float scale = 1f, bool useLighting = true, bool headOnly = false) {
+ 			if (drawPlayer.ShouldNotDraw)
+ 				return;
+ 
+@@ -115,79 +_,116 @@
+ 			_drawData.Clear();
+ 			_dust.Clear();
+ 			_gore.Clear();
+-			drawinfo.BoringSetup(drawPlayer, _drawData, _dust, _gore, position, shadow, rotation, rotationOrigin);
+-			PlayerDrawLayers.DrawPlayer_extra_TorsoPlus(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_01_BackHair(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_01_2_JimsCloak(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_01_3_BackHead(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_extra_TorsoMinus(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_02_MountBehindPlayer(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_03_Carpet(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_03_PortableStool(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_extra_TorsoPlus(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_04_ElectrifiedDebuffBack(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_05_ForbiddenSetRing(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_05_2_SafemanSun(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_06_WebbedDebuffBack(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_07_LeinforsHairShampoo(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_extra_TorsoMinus(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_08_Backpacks(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_09_BackAc(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_10_Wings(ref drawinfo);
+-			PlayerDrawLayers.DrawPlayer_11_Balloons(ref drawinfo);
++			drawinfo.BoringSetup(drawPlayer, _drawData, _dust, _gore, position, shadow, rotation, rotationOrigin, useLighting);
++
++			var vanillaLayers = new List<PlayerDrawLayer> {
++				PlayerDrawLayer.TorsoIndent, //{
++				PlayerDrawLayer.HairBack,
++				PlayerDrawLayer.JimsCloak,
++				PlayerDrawLayer.HeadBack,
++				PlayerDrawLayer.TorsoUnindent, //}
++				PlayerDrawLayer.MountBack,
++				PlayerDrawLayer.Carpet,
++				PlayerDrawLayer.PortableStool,
++				PlayerDrawLayer.TorsoIndent, //{
++				PlayerDrawLayer.ElectrifiedDebuffBack,
++				PlayerDrawLayer.ForbiddenSetRing,
++				PlayerDrawLayer.SafemanSun,
++				PlayerDrawLayer.WebbedDebuffBack,
++				PlayerDrawLayer.LeinforsHairShampoo,
++				PlayerDrawLayer.TorsoUnindent, //}
++				PlayerDrawLayer.Backpacks,
++				PlayerDrawLayer.BackAcc,
++				PlayerDrawLayer.Wings,
++				PlayerDrawLayer.BalloonAcc,
++			};
++
+ 			if (drawinfo.weaponDrawOrder == WeaponDrawOrder.BehindBackArm)
+-				PlayerDrawLayers.DrawPlayer_27_HeldItem(ref drawinfo);
++				vanillaLayers.Add(PlayerDrawLayer.HeldItem);
+ 
+-			PlayerDrawLayers.DrawPlayer_12_Skin(ref drawinfo);
++			vanillaLayers.Add(PlayerDrawLayer.Skin);
+ 			if (drawinfo.drawPlayer.wearsRobe && drawinfo.drawPlayer.body != 166) {
```

Comment:
maybe all vanilla layers could be automatically added all the time, and we just set visibility in here?

Reply:
They're already always added. The conditions only affect their ordering.